src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php

```diff
@@ -163,7 +164,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         $stopsWhen = [];
-        if ($limit = $input->getOption('limit')) {
+        if (null !== ($limit = $input->getOption('limit'))) {
+            if (!is_numeric($limit) || 0 >= $limit) {
+                throw new InvalidOptionException(sprintf('Limit "%s" is not a positive integer.', $limit));
```

Comment:
In some places, we have this wording
 `throw new InvalidArgumentException(sprintf('Argument $maxItems must be a positive integer, %d passed.', $maxItems));`

```suggestion
                throw new InvalidOptionException(sprintf('Option "limit" must be a positive integer, "%s" passed.', $limit));
```